File: src/main/java/com/linebot/batch/NoticeGarbageOutBatch.java
package com.linebot.batch;

import com.linebot.service.notice.NoticeGarbageOutService;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.annotation.Profile;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Controller;

@Slf4j
@AllArgsConstructor
@Controller
@Profile({"local", "develop"})
public class NoticeGarbageOutBatch {
	private NoticeGarbageOutService noticeGarbageOutService;

	@Scheduled(cron = "0 20 21 * * 4", zone = "Asia/Tokyo")
	public void execute() {
		noticeGarbageOutService.executeDayBefore();
	}
}

File: src/main/kotlin/com/linebot/batch/NoticeGarbageOutBatch.kt
package com.linebot.batch

import com.linebot.service.notice.NoticeGarbageOutService
import org.springframework.context.annotation.Profile
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Controller

@Controller
@Profile("release", "local", "develop")
class NoticeGarbageOutBatch(
    private val noticeGarbageOutService: NoticeGarbageOutService
) {

    @Scheduled(cron = "0 20 21 * * 4", zone = "Asia/Tokyo")
    fun execute() {
        noticeGarbageOutService.executeDayBefore()
    }
}
